Source/lighting.cpp

```diff
@@ -361,14 +365,19 @@ void MakeLightTable()
 				lightTable[idx] = color;
 			}
 		}
+		FullyLitLightTable = nullptr; // A color map is used for the ceiling animation, so even fully lit tiles have a color map
 	} else if (IsAnyOf(leveltype, DTYPE_NEST, DTYPE_CRYPT)) {
 		// Make the lava fully bright
 		for (auto &lightTable : LightTables)
 			std::iota(lightTable.begin(), lightTable.begin() + 16, uint8_t { 0 });
 		LightTables[15][0] = 0;
 		std::fill_n(LightTables[15].begin() + 1, 15, 1);
+		FullyDarkLightTable = nullptr; // Tiles in Hellfire levels are never completely black
 	}
 
+	assert((FullyLitLightTable != nullptr) == (LightTables[0][0] == 0 && std::adjacent_find(LightTables[0].begin(), LightTables[0].end() - 1, [](auto x, auto y) { return (x + 1) != y; }) == LightTables[0].end() - 1));

```

Comment:
Can you please add a comment about what these asserts are testing

Reply:
I opened #7168 for it 🙂